src/ui/public/timepicker/timepicker.js

```diff
@@ -76,11 +76,19 @@ module.directive('kbnTimepicker', function (quickRanges, timeUnits, refreshInter
       });
 
       $scope.$watch('absolute.from', function (date) {
-        if (_.isDate(date)) $scope.absolute.from = moment(date);
+        // If it is a date, it was set from the timepicker, so we set it to the start of day and transform it into a moment
+        if (_.isDate(date)) {
+          date.setHours(0, 0, 0, 0); // Start of day
```

Comment:
I had this thought too, but I don't think the version of the date picker we're using supports `ngModelOptions`. I guess I should look into upgrading the library.